src/test/java/com/meli/hseixas/listaspartidasfutebol/service/PartidaServiceTest.java

```diff
@@ -0,0 +1,434 @@
+        when(partidaRepository.findAll()).thenReturn(partidas);
+
+        assertThrows(ResponseStatusException.class, () -> partidaService.listarTodasAsPartidas());
+    }
+
+//    TESTES DO MÉTODO listarPartidaPorId()
+    @Test
+    void testListarPartidaPorId_QuandoExistePartida() {
+        addPartidas();
+        Partida partidaEsperada = partidas.get(0);
+
+        when(partidaRepository.findById(partidaEsperada.getId())).thenReturn(Optional.of(partidaEsperada));
+
+        PartidaDto partidaDto = partidaService.listarPartidaPorId(partidaEsperada.getId());
+
+        assertNotNull(partidaDto);
+        assertEquals(partidaEsperada.getId(), partidaDto.getId());
+    }
+
+    @Test
+    void testListarPartidaPorId_QuandoNaoExistePartida() {
+        when(partidaRepository.findById(ID_PARTIDA_INEXISTENTE)).thenReturn(Optional.empty());
+
+        assertThrows(ResponseStatusException.class, () -> partidaService.listarPartidaPorId(ID_PARTIDA_INEXISTENTE));
+    }
+
+//    TESTES DO MÉTODO listarGoleadas()
+    @Test
+    void testListarGoleadas_QuandoExistemGoleadas() {
+        addPartidas();
+        partidas.add(new Partida(4L, "Barra Mansa", "Bangu", 4, 1, LocalDateTime.of(2022, 10, 12, 20, 30, 00), "Leão do Sul"));
+        partidas.add(new Partida(5L, "Volta Redonda", "Madureira", 3, 0, LocalDateTime.of(2022, 10, 12, 20, 30, 00), "Raulino de Oliveira"));
+
+        when(partidaRepository.findAll()).thenReturn(partidas);
+
+        List<PartidaDto> goleadas = partidaService.listarGoleadas();
+
+        for (PartidaDto partidaDto : goleadas){
+            assertTrue(Math.abs(partidaDto.getPlacarMandante() - partidaDto.getPlacarVisitante()) >= 3);
+        }
+        assertEquals(2, goleadas.size());
+    }
+
+    @Test
+    void testListarGoleadas_QuandoNaoExistemGoleadas() {
+        addPartidas();
+
+        when(partidaRepository.findAll()).thenReturn(partidas);
+
+        List<PartidaDto> goleadas = partidaService.listarGoleadas();
+
+        assertTrue(goleadas.isEmpty());
+    }
+
+    @Test
+    void testListarGoleadas_QuandoNaoExistemPartidas() {
+        when(partidaRepository.findAll()).thenReturn(partidas);
+
+        assertThrows(ResponseStatusException.class, () -> partidaService.listarGoleadas());
+    }
+
+//    TESTES DO MÉTODO listarZeroGols()
+    @Test
+    void testListarZeroGols_QuandoExistemZeroGols() {
+        addPartidas();
+        partidas.add(new Partida(4L, "Barra Mansa", "Bangu", 0, 0, LocalDateTime.of(2022, 10, 12, 20, 30, 00), "Leão do Sul"));
+        partidas.add(new Partida(5L, "Volta Redonda", "Madureira", 0, 0, LocalDateTime.of(2022, 10, 12, 20, 30, 00), "Raulino de Oliveira"));
+
+        when(partidaRepository.findAll()).thenReturn(partidas);
+
+        List<PartidaDto> zeroGols = partidaService.listarZeroGols();
+
+        for (PartidaDto partidaDto : zeroGols){
+            assertTrue(partidaDto.getPlacarMandante() == 0 && partidaDto.getPlacarVisitante() == 0);
```

Comment:
assert 0 fica mais clean

```suggestion
            assertEquals(0, partidaDto.getPlacarMandante());
            assertEquals(0, partidaDto.getPlacarVisitante());
```

Reply:
Feito!